src/vm/runners/builtin_runner/ec_op.rs

```diff
@@ -609,4 +620,29 @@ mod tests {
             ))
         );
     }
+
+    #[test]
+    fn get_memory_accesses() {
+        let builtin = EcOpBuiltinRunner::new(256);
+        let mut vm = vm!();
+
+        assert_eq!(
+            builtin.get_memory_accesses(&vm),
+            Err(MemoryError::MissingSegmentUsedSizes),
+        );
+
+        vm.segments.segment_used_sizes = Some(vec![0]);
+        assert_eq!(builtin.get_memory_accesses(&vm), Ok(vec![]));
+
+        vm.segments.segment_used_sizes = Some(vec![4]);
+        assert_eq!(
+            builtin.get_memory_accesses(&vm),
+            Ok(vec![
+                (builtin.base, 0).into(),
+                (builtin.base, 1).into(),
+                (builtin.base, 2).into(),
+                (builtin.base, 3).into(),
+            ]),
+        );
+    }
```

Comment:
Please separate these into different tests, as they test different cases

Reply:
What's the criteria for deciding when to split a test or not?

Reply:
Tests are meant to try specific cases, so if we make a change and a test fails we can know specifically which case is the one failing. For example here, there are 3 different cases, the first one doesn't have used_sizes, the second one has an empty vec for used_sizes and the third one has a filled used_sizes. If we were to have failures on cases 1 and 3, with this test we would only be able to see the failure in test 1, and we want as much information as possible when a functionality fails.

Reply:
I think I more or less get it now. Thanks for the explanation.